### ISSUE TITLE
question abbout audio tagging

### ISSUE THREAD

**Issue by heiko134470** [yongxuUSTC/dcase2017_task4_cvssp#2]
Hi,
you described in this paper
http://www.cs.tut.fi/sgn/arg/dcase2017/documents/challenge_technical_reports/DCASE2017_Xu_146.pdf
about the model of audio tagging as following:
-----------------------------------------------------------------------------
5. Whole framework
We treat one CNN layer followed by one batch normalization layer
and the learned gating layer as a block. Two such kind of blocks and
one max-pooling layer form a unit.  Four units were concatenated
together to form the whole system.   The final output layer is the
weighted mean along the time axis to get the 17 output nodes.
-------------------------------------------------------------------------------
I didn't exactly understand about the part of model after CNN layer or final output layer. How does it look like?
the following part is correct?
---------------------------------------------------------------------------------
    cnn1 = block(a1)
    cnn1 = block(cnn1)
    cnn1 = MaxPooling2D(pool_size=(1, 2))(cnn1)
    
    cnn1 = block(cnn1)
    cnn1 = block(cnn1)
    cnn1 = MaxPooling2D(pool_size=(1, 2))(cnn1)
    
    cnn1 = block(cnn1)
    cnn1 = block(cnn1)
    cnn1 = MaxPooling2D(pool_size=(1, 2))(cnn1)
    
    cnn1 = block(cnn1)
    cnn3 = block(cnn1)
    cnn3 = MaxPooling2D(pool_size=(1, 2))(cnn1)
    
    cnnout = Conv2D(256, (3, 3), padding="same", activation="relu", use_bias=True)(cnn3)
    cnnout = MaxPooling2D(pool_size=(1, 4))(cnnout)
    
    cnnout = Reshape((240, 256))(cnnout)   # Time step is downsampled to 30.     
    out = TimeDistributed(Dense(num_classes, activation='sigmoid'), name='localization_layer')(out)
    det =TimeDistributed(Dense(num_classes, activation='softmax'))(out)
    out=Multiply()([out,det])
    out=Lambda(outfunc, output_shape=(num_classes,))([out, det])
    
    model = Model(input_logmel, out)
    
---------------------------------------------------------------------------------------------------

**Comment by qiuqiangkong** [comment]
Hi,


    many thanks for your interest!


    After the final convolutional layer, we have an classification part "out" and an attention part "det". The final result is the weighted sum of "out" and "det". This is what the Lambda function do.


    Please let us know if there are further questions!


Best wishes,


Qiuqiang

________________________________
